test/lsp-io-tests.el

```diff
@@ -58,5 +58,18 @@
 		(should (equal (lsp--parser-read p "\"somedata\":2}")
 									 '("{\"somedata\":2}")))))
 
+(ert-deftest lsp--parser-read--multiple-chunks-ignored ()
+  (let* ((log '())
+         (p (make-lsp--parser :workspace lsp--test-workspace))
+         (lsp--parser-reset p)
+         (filter (lsp--parser-make-filter p '("IGNORE_ME"))))
+    (cl-letf (((symbol-function 'lsp--parser-on-message)
+               (lambda (p msg) (push msg log))))
```

Comment:
This is totally fine, but you could also use a mock framework such as `el-mock`:

```elisp
(with-mock
  (mock (lsp--parser-on-message * "{}") :times 2)
   ...)
```